vortex-array/src/stats/mod.rs

```diff
@@ -52,12 +52,14 @@ pub trait Statistics {
     /// Computes the value of the stat if it's not present
     fn compute(&self, stat: Stat) -> Option<Scalar>;
 
+    /// Applies the given function to the statistic if it's present
     fn with_stat_value<'a>(
```

Comment:
@robert3005 what are these for?

Reply:
These are so we can do `&Scalar -> T` instead of doing `Scalar -> T`